### ISSUE TITLE
"Welcome to Test Pilot!" pop-up is wrongly dismissed when any action is performed

### ISSUE THREAD

**Issue by SoftVision-CosminMuntean** [mozilla/testpilot#2997]
[Notes]:
- It seems that after installing Test Pilot add-on, the page is wrongly refreshed when any action is performed (clicking anywhere on the page or focusing other appreciations).

[Affected versions]:
- Firefox 56.0 and above

[Affected Platforms]:
- All Windows
- All Mac OS
- All Linux

[Prerequisites]:
- Following 5 prefs have been set or added on a new profile: "extensions.legacy.enabled = true" "xpinstall.signatures.required = false" "testpilot.env = stage" "extensions.webapi.testing = true" "extensions.install.requireBuiltInCerts = false".

[Steps to reproduce]:
1. Open Firefox browser with the profile from prerequisites and navigate to https://testpilot.stage.mozaws.net page.
2. Chose to install Test Pilot add-on.
3. Click on the email textbox from the "Welcome to Test Pilot" pop-up.
4. Observe the broser behavior.

[Expected result]:
- The email textbox is focused and the writing pointer is displayed.

[Actual result]:
- The Test Pilot page is refreshed and the pop-up is dismissed.

[Additional Notes]:
- The issue is not reproducible on Test Pilot production server.
- Here is a screen recording with the issue:
![welcome](https://user-images.githubusercontent.com/19226534/32164954-ab5418aa-bd69-11e7-859e-7b41a6c8faef.gif)

**Comment by lmorchard** [comment]
Will take a look today - *might* be related to #2983 that I just merged (since it dealt with dialogs) but I didn't see this issue when I reviewed the PR.

**Comment by lmorchard** [comment]
Huh, this is weird: I can reproduce on [staging](https://testpilot.stage.mozaws.net/experiments) and [dev](https://testpilot.dev.mozaws.net/experiments), but not in my local environment. I don't see any apparently relevant errors in the console. Still investigating...

**Comment by kgeorgiou** [comment]
@lmorchard, looking into it as well.

**Comment by lmorchard** [comment]
Another interesting detail: I could finally reproduce on my local machine after rebuilding the add-on. So, I'm thinking maybe this has more to do with the add-on than the welcome dialog

**Comment by kgeorgiou** [comment]
I've managed to narrow it down to: https://github.com/mozilla/testpilot/blob/856b774168ce10d704b63602934f95ff4921fb44/frontend/src/app/lib/InstallManager.js#L18-L38 that was changed in #2984. Reverting the changes in [InstallManager.js](https://github.com/mozilla/testpilot/blob/master/frontend/src/app/lib/InstallManager.js) seems to produce the normal/expected behaviour.
I'm not familiar with how the addon install manager is supposed to work, but the promise https://github.com/mozilla/testpilot/blob/856b774168ce10d704b63602934f95ff4921fb44/frontend/src/app/lib/InstallManager.js#L23 is _resolved_ prematurely when any action is performed, as Cosmin described above.

**Comment by kgeorgiou** [comment]
I've tried putting something together. 
There is an issue where both the Test Pilot and the Experiment addons invoke the following function when installed:
https://github.com/mozilla/testpilot/blob/856b774168ce10d704b63602934f95ff4921fb44/frontend/src/app/lib/InstallManager.js#L18-L21
The URL for the Test Pilot addon is different from the experiments. There's no `files/` or `@` to be found and the `experimentTitle` ends up wrongly being "https". There should be a way to split this logic.

Also regarding the states when an Experiment is installed; notice on the staging server:
1. Enable experiment.
2. Click "Add" on the Permissions "Cancel / Add" pop-up. (Triggers the 'onInstallEnded' event that the `install` instance used to listen to.)
3. Click "OK" on the new pop-up informing us that the experiment is installed. (Resolves `install().then()`)
4. *Then* the tour dialog pops-up

Whereas in production the tour dialogs show up right after Step 2 (i.e. 'onInstallEnded' instead on `install().then()`. ) I followed the pattern that's currently in production in my commit above. Not sure which one is the intended behaviour, I can update accordingly and open a PR.

**Comment by fzzzy** [comment]
@kgeorgiou That patch looks good. I think "experimentTitle ends up being https" is an unrelated bug, can you file a new bug for that, and open a new pull request?

While your fix for this bug does work, we would like to have a fix that does not rely on onInstallEnded. I believe I have a patch that works now, which is removing pollMainAddonAvailability which was old code that is no longer required, but is causing this bug.

**Comment by SoftVision-CosminMuntean** [comment]
I have retested this issue on Test Pilot dev and stage servers and is no longer reproducible. Marking as verified.
